core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -314,31 +319,51 @@ class ReassignPartitionsCommand(zkUtils: ZkUtils, partitions: Map[TopicAndPartit
     }
   }
 
-  def addThrottledReplicaList(): Unit = {
-    //apply the throttle to all move destinations and all move sources
-    val existing = zkUtils.getReplicaAssignmentForTopics(partitions.map(_._1.topic).toSeq)
-    val moves = replicaMoves(existing, proposed = partitions)
-    for (topic <- partitions.keySet.map(tp => tp.topic).toSeq.distinct)
-      AdminUtils.changeTopicConfig(zkUtils, topic, new Properties {
-        put(LogConfig.ThrottledReplicasListProp, moves.get(topic).get )
-      })
-    println(s"Throttles were added to the following replicas: $moves")
+  def assignThrottledReplicas(): Unit = {
+    val topicsForMove = partitions.map { case (tp, _) => tp.topic }.toSeq
+    val current = zkUtils.getReplicaAssignmentForTopics(topicsForMove)
+    assignThrottledReplicas(current, partitions)
+  }
+
+  private[admin] def assignThrottledReplicas(allExisting: Map[TopicAndPartition, Seq[Int]], allProposed: Map[TopicAndPartition, Seq[Int]], admin: AdminUtilities = AdminUtils): Unit = {
+    for (topic <- allProposed.keySet.map(tp => tp.topic).toSeq.distinct) {
+      val (existing, proposed) = filterBy(topic, allExisting, allProposed)
+
+      //Apply the leader throttle to all replicas that exist before the re-balance.
+      val leader = format(preRebalanceReplicaForMovingPartitions(existing, proposed))
+
+      //Apply a follower throttle to all "move destinations".
+      val follower = format(postRebalanceReplicasThatMoved(existing, proposed))
+
+      admin.changeTopicConfig(zkUtils, topic, wrap((LeaderThrottledReplicasListProp, leader), (FollowerThrottledReplicasListProp, follower)))
+      info(s"Updated leader-throttled replicas for topic $topic with: $leader")
+      info(s"Updated follower-throttled replicas for topic $topic with: $follower")
+    }
+  }
+
+  private def postRebalanceReplicasThatMoved(existing: Map[TopicAndPartition, Seq[Int]], proposed: Map[TopicAndPartition, Seq[Int]]): Map[TopicAndPartition, Seq[Int]] = {
+    //For each partition in the proposed list, filter out any replicas that exist now (i.e. not moving)
+    existing.map { case (tp, current) =>
+      (tp -> proposed.get(tp).get.filterNot(current.toSet))
```

Comment:
I've added validation to ensure that all proposed partitions exist in the cluster (added test too)